src/ddscxx/include/org/eclipse/cyclonedds/topic/datatopic.hpp

```diff
@@ -647,9 +647,21 @@ ddsi_serdata * serdata_from_iox_buffer(
     }
 
     // key handling
-    const auto& msg = *static_cast<const T*>(d->iox_chunk);
-    d->key_md5_hashed() = to_key(msg, d->key());
-    d->populate_hash();
+    if (typecmn->fixed_size == 1) {
```

Comment:
Nitpick:
```suggestion
    if (typecmn->fixed_size) {
```
(or, alternatively, `typecmn->fixed_size != 0`)

What you wrote is perfectly correct given that `fixed_size` is a single bit, but it seems a possible risk if anything gets refactored.

Reply:
hi @eboasson Your change is great and it will make codes more robust. Thanks~